pkg/kubelet/dockertools/manager.go

```diff
@@ -2082,3 +2082,14 @@ func (dm *DockerManager) GetNetNs(containerID kubecontainer.ContainerID) (string
 func (dm *DockerManager) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy) error {
 	return dm.containerGC.GarbageCollect(gcPolicy)
 }
+
+// Ping attempts to connect to the docker daemon and determine its health.
+// `docker ps` has been very useful for determining the state of the daemon.
+// Issue #10959 has more information on this rationale.
+func (dm *DockerManager) Ping() error {
+	_, err := dm.client.ListContainers(docker.ListContainersOptions{All: true})
```

Comment:
Are we sure we want to re-introduce this heavy operation for docker healthchecking periodically (every 30s)?

I used to ping docker socket every 2 mins through monit, but you talked us off from that when replacing monit with supervisord. Shouldn't this one is way more heavier than previous version?